client_token_gateway.go

```diff
@@ -7,23 +7,23 @@ import (
 
 const clientTokenVersion = 2
 
+// ClientTokenGateway represents the provider client token generation
 type ClientTokenGateway struct {
 	*Braintree
 }
 
 // Generate generates a new client token.
-func (g *ClientTokenGateway) Generate(ctx context.Context) (string, error) {
-	return g.generate(ctx, &ClientTokenRequest{
-		Version: clientTokenVersion,
-	})
-}
+func (g *ClientTokenGateway) Generate(ctx context.Context, req *ClientTokenRequest) (string, error) {
```

Comment:
I think in the case of client token generation it's worth leaving the existing functions as they are and adding a new function `GenerateWithRequest` that exposes the full variation of options. The function hasn't exposed the additional fields and options until now so we can make a guess that existing users of the package don't need the additional functionality and we should avoid inconveniencing them by breaking the API.